pkg/virt-controller/watch/vm.go

```diff
@@ -2718,25 +2707,27 @@ func (c *VMController) applyDevicePreferences(vm *virtv1.VirtualMachine, vmi *vi
 	return nil, nil
 }
 
-func (c *VMController) handleDynamicInterfaceRequests(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) error {
-	if len(vm.Status.InterfaceRequests) == 0 {
+func (c *VMController) handleDynamicInterfaces(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) error {
+	if vmi == nil || vmi.DeletionTimestamp != nil {
 		return nil
 	}
 
-	vmTemplateCopy := vm.Spec.Template.Spec.DeepCopy()
-	for i := range vm.Status.InterfaceRequests {
-		vmTemplateCopy = controller.ApplyNetworkInterfaceRequestOnVMISpec(
-			vmTemplateCopy, &vm.Status.InterfaceRequests[i])
-	}
-	vm.Spec.Template.Spec = *vmTemplateCopy
-
-	if vmi != nil && vmi.DeletionTimestamp == nil {
-		vmiSpecCopy := vmi.Spec.DeepCopy()
-		for i := range vm.Status.InterfaceRequests {
-			vmiSpecCopy = controller.ApplyNetworkInterfaceRequestOnVMISpec(
-				vmiSpecCopy, &vm.Status.InterfaceRequests[i])
+	vmiSpecCopy := vmi.Spec.DeepCopy()
+	vmiIndexedInterfaces := vmispec.IndexInterfaceSpecByName(vmiSpecCopy.Domain.Devices.Interfaces)
+	vmIndexedNetworks := vmispec.IndexNetworkSpecByName(vm.Spec.Template.Spec.Networks)
+	for _, vmIface := range vm.Spec.Template.Spec.Domain.Devices.Interfaces {
+		_, existsInVMISpec := vmiIndexedInterfaces[vmIface.Name]
+		if !existsInVMISpec && vmIface.State != virtv1.InterfaceStateAbsent && vmIface.InterfaceBindingMethod.Bridge != nil {
```

Comment:
Not a must, but consider wrapping these so it will be clear what it is used for.
E.g. `shouldBeHotPlug`.

Same for the one after: `shouldBeHotUnplug`.